packages/core-data/src/types.ts

```diff
@@ -0,0 +1,85 @@
+export type Context = 'edit' | 'embed' | 'view';
+
+interface Edit {}
+
+// @ts-ignore
+export interface EntityRecord< C extends Context > {}
+
+export interface EntityConfig< Record extends EntityRecord<any> > {
+	/** Path in WP REST API from which to request records of this entity. */
+	baseURL: string;
+
+	/** Arguments to supply by default to API requests for records of this entity. */
+	baseURLParams: { context: Context };
```

Comment:
The type here is more like

```suggestion
	baseURLParams?: Record<string, string> & { context?: Context }
```